pkg/secrets/pushtofile/secrets_groups_test.go

```diff
@@ -0,0 +1,170 @@
+					FilePerms:    DEFAULT_FILE_PERMS,
+					FileTemplate: "",
+				},
+			},
+		),
+	},
+	{
+		description: "invalid file path",
+		contents: map[string]string{
+			"conjur.org/conjur-secrets.cache":   someValidSecrets,
+			"conjur.org/secret-file-path.cache": "/this/absolute/path",
+		},
+		assert: assertProperError("must contain relative path"),
+	},
+	{
+		description: "invalid file path with template specified",
+		contents: map[string]string{
+			"conjur.org/conjur-secrets.cache":       someValidSecrets,
+			"conjur.org/secret-file-path.cache":     "this/relative/directory/",
+			"conjur.org/secret-file-template.cache": someValidTemplate,
+		},
+		assert: assertProperError("contains directory instead of file"),
+	},
+}
+
+func TestExtractSecretsGroupsFromAnnotations(t *testing.T) {
+	for _, tc := range extractSecretsGroupsTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			secretGroups, err := ExtractSecretsGroupsFromAnnotations(tc.contents)
+
+			tc.assert(t, secretGroups, err)
+		})
+	}
+}
+
+func TestExtractFileFormatFromAnnotations(t *testing.T) {
+	type fileFormatTestCase struct {
+		input       string
+		output      SecretsFileFormat
+		valid       bool
+		hasTemplate bool
+	}
+
+	fileFormatTestCases := []fileFormatTestCase{
+		{input: "", output: SecretsFileFormat(FILE_FORMAT_YAML), valid: true, hasTemplate: false},
+		{input: "YAML", output: SecretsFileFormat(FILE_FORMAT_YAML), valid: true, hasTemplate: false},
+		{input: "yaml", output: SecretsFileFormat(FILE_FORMAT_YAML), valid: true, hasTemplate: false},
+		{input: "JSON", output: SecretsFileFormat(FILE_FORMAT_JSON), valid: true, hasTemplate: false},
+		{input: "json", output: SecretsFileFormat(FILE_FORMAT_JSON), valid: true, hasTemplate: false},
+		{input: "DOTENV", output: SecretsFileFormat(FILE_FORMAT_DOTENV), valid: true, hasTemplate: false},
+		{input: "dotenv", output: SecretsFileFormat(FILE_FORMAT_DOTENV), valid: true, hasTemplate: false},
+		{input: "BASH", output: SecretsFileFormat(FILE_FORMAT_BASH), valid: true, hasTemplate: false},
+		{input: "bash", output: SecretsFileFormat(FILE_FORMAT_BASH), valid: true, hasTemplate: false},
+		{input: "PLAINTEXT", output: SecretsFileFormat(FILE_FORMAT_PLAINTEXT), valid: true, hasTemplate: false},
+		{input: "plaintext", output: SecretsFileFormat(FILE_FORMAT_PLAINTEXT), valid: true, hasTemplate: false},
+		{input: "invalid", output: SecretsFileFormat(FILE_FORMAT_NONE), valid: false, hasTemplate: false},
+		{input: "dotenv", output: SecretsFileFormat(FILE_FORMAT_NONE), valid: true, hasTemplate: true},
+		{input: "invalid", output: SecretsFileFormat(FILE_FORMAT_NONE), valid: false, hasTemplate: true},
+	}
+
+	secretsGroupTestCase := extractSecretsGroupsTestCase{
+		description: "file type annotation",
+		contents: map[string]string{
+			"conjur.org/conjur-secrets.cache":     someValidSecrets,
+			"conjur.org/secret-file-format.cache": "",
+			"conjur.org/secret-file-path.cache":   "this/relative/file",
```

Comment:
A file needs to be specified here since some of the file format tests test with a valid template, which requires a file, not a directory (which is the default)